### PR TITLE
expand pool once pool is full and ingest continues after pool expand completes

### PR DESCRIPTION
**Pull request by vprabhakar-px**
https://portworx.atlassian.net/browse/PTX-18927
Fill volumes completely , then resize volume by 50%, verify IO on volumes in Longevity

https://jenkins.pwx.dev.purestorage.com/view/CLOUD-DRIVE-3.0.0/job/Torpedo/job/tp-fa-cd-3-0-Centos/175/console

https://aetos.pwx.purestorage.com/resultSet/testSetID/270649

### CODE REVIEW

**Comment by vprabhakar-px** [comment]
STEP: destroy the vdbench-heavyload app's volumes
2023-07-18 16:54:08 +0530:[INFO] [{ResizeVolumeAfterFull}] [tests.DeleteVolumes.func1:#1547] - destroy the vdbench-heavyload app's volumes
2023-07-18 16:54:08 +0530:[INFO] [{ResizeVolumeAfterFull}] [k8s.(*K8s).DeleteVolumes:#3848] - [vdbench-heavyload] Destroyed storage class: vdbench-sc
2023-07-18 16:54:08 +0530:[INFO] [{ResizeVolumeAfterFull}] [k8s.(*K8s).DeleteVolumes:#3854] - [vdbench-heavyload] PVC is not found: vdbench-pvc, skipping deletion
2023-07-18 16:54:08 +0530:[INFO] [{ResizeVolumeAfterFull}] [k8s.(*K8s).DeleteVolumes:#3854] - [vdbench-heavyload] PVC is not found: vdbench-pvc-output, skipping deletion
INFO[2023-07-18 16:54:09] Verifying : Description : Test completed successfully ? 
INFO[2023-07-18 16:54:09] Actual:PASS, Expected: PASS         
INFO[2023-07-18 16:54:10] --------Test End------                       
INFO[2023-07-18 16:54:10] #Test: ResizeVolumeAfterFull                 
INFO[2023-07-18 16:54:10] #Description: Fill volumes completely , then resize volume by 50%, verify IO on volumes in Longevity  
INFO[2023-07-18 16:54:10] ------------------------                     

• [SLOW TEST:503.650 seconds]
{ResizeVolumeAfterFull}
/Users/vprabhakar/Desktop/work/torpedo/torpedo/torpedo/tests/basic/storage_pool_test.go:9766
  Fill volumes completely , then resize volume by 50%, verify IO on volumes in Longevity
  /Users/vprabhakar/Desktop/work/torpedo/torpedo/torpedo/tests/basic/storage_pool_test.go:9779

File: tests/common.go [hunk 1]
```diff
@@ -7581,3 +7581,182 @@ func GetClusterProviders() []string {
...
+	num, err := strconv.ParseUint(output, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+
+	return num, nil
+}
+
+// IsIORunningOnVolume return true if IO is running on the volume
+func IsIORunningOnVolume(vol volume.Volume) (bool, error) {
```
Comment (lsrinivas-pure) [inline tests/common.go:7649]: can you reuse this IsIOsInProgressForTheVolume?

File: tests/common.go [hunk 2]
```diff
@@ -7581,3 +7581,182 @@ func GetClusterProviders() []string {
...
+
+	for _, v := range poolsStatus {
+		if v == "Offline" {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func GetPoolUuidsWithStorageFull() ([]string, error) {
```
Comment (lsrinivas-pure) [inline tests/common.go:7599]: delete this if not required

File: tests/common.go [hunk 3]
```diff
@@ -7581,3 +7581,182 @@ func GetClusterProviders() []string {
...
+
+	log.Infof("Total size of Volume is [%v] and used size is [%v]", totalSize, usedSize)
+	percentageFull := calculatePercentage(float64(usedSize), float64(totalSize))
+	log.Infof("Volume [%v] : Percentage of size consumed [%v]", vol.Name, percentageFull)
+
+	return percentageFull, nil
+}
+
+// IsStorageNodeDown returns true if storage node is down
+func IsStorageNodeDown(n node.Node) (bool, error) {
```
Comment (lsrinivas-pure) [inline tests/common.go:7733]: can you please reuse waitForStorageDown

File: tests/common.go [hunk 4]
```diff
@@ -7581,3 +7581,182 @@ func GetClusterProviders() []string {
...
+	}
+	if *nodeStatus == opsapi.Status_STATUS_STORAGE_DOWN {
+		log.InfoD("Status of the node is [%v]", nodeStatus)
+		return true, nil
+	}
+	return false, nil
+}
+
+// GetPoolCapacityUsed Get Pool capacity percentage used
+func GetPoolCapacityUsed(poolUUID string) (float64, error) {
```
Comment (lsrinivas-pure) [inline tests/common.go:7746]: delete if not required

File: tests/common.go [hunk 5]
```diff
@@ -7581,3 +7581,182 @@ func GetClusterProviders() []string {
 	}
 	return clusterProviders
 }
+
+func IsPoolOffline(n node.Node) (bool, error) {
```
Comment (lsrinivas-pure) [inline tests/common.go:7585]: delete if not required

File: tests/common.go (hunk 4 above)
Comment (vprabhakar-px) [reply tests/common.go:7746]: this is part of library , but not currently in use. this is to get the fill percentage of the pool .

File: tests/common.go (hunk 5 above)
Comment (vprabhakar-px) [reply tests/common.go:7585]: function is being used to check 2 things , if the pod percentage is 90% full or pod is offline , both the cases function GetPoolUuidsWithStorageFull returns true , will keep it for now . will optimise the funciton in future

File: tests/common.go (hunk 3 above)
Comment (vprabhakar-px) [reply tests/common.go:7733]: deleted the function

File: tests/common.go (hunk 1 above)
Comment (vprabhakar-px) [reply tests/common.go:7649]: fixed

File: tests/common.go (hunk 2 above)
Comment (vprabhakar-px) [reply tests/common.go:7599]: keeping this function for now , will delete or modify the test later

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Pull request merged**